pdns/dnsdist-console.cc

```diff
@@ -45,6 +46,25 @@ void feedConfigDelta(const std::string& line)
   g_confDelta.push_back({now,line});
 }
 
+string historyFile(const bool &ignoreHOME = false)
+{
+  string ret;
+
+  struct passwd pwd;
+  struct passwd *result;
+  char buf[16384];
+  getpwuid_r(getuid(), &pwd, buf, sizeof(buf), &result);
```

Comment:
I'm wondering if we should use `geteuid()` instead, in the unlikely event `dnsdist` is started with `-u` while in foreground mode.